### PR TITLE
gasPrice suggestions for metamask and others

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

This adds the `gasPrice` parameter to non-relayer transactions in `_txHelper`.  I'm not entirely sold on the non-metamask portion of this since it always adds an extra call.  I'd like other peoples' opinion on this.

CC: @micahalcorn 

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: packages/graphql/src/mutations/_txHelper.js [hunk 1]
```diff
@@ -46,7 +50,11 @@ function useRelayer({ mutation, value }) {
   if (!contracts.config.relayer) reason = 'relayer not configured'
   if (!mutation) reason = 'no mutation specified'
 
-  if (mutation === 'makeOffer' && value && value !== '0') {
+  if (
+    ['makeOffer', 'swapAndMakeOffer'].includes(mutation) &&
```
Comment (franckc) [inline packages/graphql/src/mutations/_txHelper.js]: So prior to this we were using the relayer for swapAndMakeOffer ? That's odd, I don't remember my offers with DAI payment going thru the relayer...

File: packages/graphql/src/mutations/_txHelper.js [hunk 2]
```diff
@@ -141,6 +167,22 @@ export default function txHelper({
...
+      if (window.ethereum) {
+        // Don't use eth_gasPrice for MM
+        try {
+          gasPrice = await fetchGasPrice()
+        } catch (err) {
+          debug(`Error fetching gas price`, err)
+        }
+      } else {
+        // But use it when we're using our provider
+        gasPrice = await web3.eth.getGasPrice()
```
Comment (franckc) [inline packages/graphql/src/mutations/_txHelper.js]: Is there a point in us having an EthGasStationProvider ?
Could we get rid of it and just always set the gasPrice here ?

File: packages/graphql/src/mutations/_txHelper.js [hunk 3]
```diff
@@ -141,6 +167,22 @@ export default function txHelper({
...
+    debug(`shouldUseRelayer: ${shouldUseRelayer}`)
+
+    // Get our gas price
+    if (!shouldUseRelayer && ['mainnet', 'rinkeby'].includes(contracts.net)) {
+      if (window.ethereum) {
+        // Don't use eth_gasPrice for MM
+        try {
+          gasPrice = await fetchGasPrice()
+        } catch (err) {
+          debug(`Error fetching gas price`, err)
```
Comment (franckc) [inline packages/graphql/src/mutations/_txHelper.js]: Should we log to Sentry ?

File: packages/graphql/src/mutations/_txHelper.js (hunk 2 above)
Comment (franckc) [reply packages/graphql/src/mutations/_txHelper.js]: We could do this as a separate cleanup PR. Just asking for my own understanding... :)

File: packages/graphql/src/mutations/_txHelper.js (hunk 1 above)
Comment (mikeshultz) [reply packages/graphql/src/mutations/_txHelper.js]: No, it just so happened that it was deflected by `useProxy`.  `useProxy` didn't work for my logic so i added it here too.

File: packages/graphql/src/mutations/_txHelper.js (hunk 2 above)
Comment (mikeshultz) [reply packages/graphql/src/mutations/_txHelper.js]: `web3.eth.getGasPrice` will use the provider(if it's there).  And so will all the services that don't use GraphQL.

File: packages/graphql/src/mutations/_txHelper.js (hunk 3 above)
Comment (mikeshultz) [reply packages/graphql/src/mutations/_txHelper.js]: I don't think there's been any Sentry integration in graphql(yet).

Comment (franckc) [reply packages/graphql/src/mutations/_txHelper.js]: Ah yes, very good point. You are still very lucid at this late hour you know... :)

**Pull request merged**